### PR TITLE
feat: Make all Dungeon solver colors customizable

### PR DESCRIPTION
**Pull request by Coccocoahelper**
Not ready yet, but likely not going to break things. Try it once it compiles.

### CODE REVIEW

**Review by Sychic — CHANGES_REQUESTED** [review]
e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:20:38 Unresolved reference: config
e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:38:9 Unresolved reference: ticTacToeSolverColor

**Comment by Coccocoahelper** [comment]
> e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:20:38 Unresolved reference: config e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:38:9 Unresolved reference: ticTacToeSolverColor

My bad, fixing it.

**Comment by Coccocoahelper** [comment]
> e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:20:38 Unresolved reference: config e: file:///home/runner/work/SkytilsMod/SkytilsMod/src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt:38:9 Unresolved reference: ticTacToeSolverColor

Can you make it try to compile rn?

File: src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt [hunk 1]
```diff
@@ -35,7 +35,7 @@ fun setChestColor(
     ci: CallbackInfo
 ) {
     if (te.pos == ThreeWeirdosSolver.riddleChest) {
-        ticTacToeSolverColor.bindColor()
+        Skytils.config.threeWeirdosSolverColor.bindColor()
```
Comment (Sychic) [inline src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt]: This should just import the static and not the surrounding class.

**Review by Sychic — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt (hunk 1 above)
Comment (Coccocoahelper) [reply src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt]: I'm going to try that out, thank you for the advice! Also, are Kotlin mixins viable for other… mods?

Comment (Coccocoahelper) [reply src/main/kotlin/gg/skytils/skytilsmod/mixins/hooks/renderer/TileEntityChestRendererHook.kt]: > This should just import the static and not the surrounding class.

That doesn't appear possible. I'll revert it to the old method

**Comment by Coccocoahelper** [comment]
Doesn't compile because of Gradle dependencies (?). I need someone to take a closer look at the errors.

**Pull request merged**